gdnative-core/src/dictionary.rs

```diff
@@ -134,64 +178,219 @@ impl Dictionary {
     /// Modifying the same underlying collection while observing the safety assumptions will
     /// not violate memory safely, but may lead to surprising behavior in the iterator.
     #[inline]
-    pub fn iter(&self) -> Iter {
-        Iter::new(self)
+    pub fn iter(&self) -> IterShared {
+        IterShared::new(self)
     }
 
-    #[doc(hidden)]
+    /// Create a copy of the dictionary.
+    ///
+    /// This creates a new dictionary and is **not** a cheap reference count
+    /// increment.
     #[inline]
-    pub fn sys(&self) -> *const sys::godot_dictionary {
-        &self.0
+    pub fn duplicate(&self) -> Dictionary<Unique> {
+        let d = Dictionary::new();
+        for (k, v) in self {
+            d.insert(&k, &v);
+        }
+        d
     }
+}
```

Comment:
Is there anything preventing this from being a common method? I think the `Unique` version looks the same.

Reply:
Yes, the iterators are different types and since the bound of `Access` is for all `ThreadAccess` no exhaustive check can be made, so the `IntoIterator` can't be expressed for all `Dictionary<T>` types, so they have to be hardcoded. Maybe with a blanket implementation this can be solved, I could try that if this is something you would rather see gone (or documented).

Reply:
Understood. This sounds like something #413 can help solve in the future, since then we can write an iterator type that is generic over anything we can get a `godot_dictionary` out of. Let's keep it this way for now.